store/postgres/src/dynds/private.rs

```diff
@@ -206,37 +207,80 @@ impl DataSourcesTable {
         conn: &PgConnection,
         dst: &DataSourcesTable,
         target_block: BlockNumber,
+        src_manifest_idx_and_name: &[(i32, String)],
+        dst_manifest_idx_and_name: &[(i32, String)],
     ) -> Result<usize, StoreError> {
         // Check if there are any data sources for dst which indicates we already copied
         let count = dst.table.clone().count().get_result::<i64>(conn)?;
         if count > 0 {
             return Ok(count as usize);
         }
 
-        let query = format!(
-            "\
-            insert into {dst}(block_range, causality_region, manifest_idx, parent, id, param, context)
-            select case
-                when upper(e.block_range) <= $1 then e.block_range
-                else int4range(lower(e.block_range), null)
+        type Tuple = (
+            (Bound<i32>, Bound<i32>),
+            i32,
+            Option<Vec<u8>>,
+            Option<serde_json::Value>,
+            i32,
+        );
+
+        let src_tuples = self
+            .table
+            .clone()
+            .filter(diesel::dsl::sql("lower(block_range) <= ").bind::<Integer, _>(target_block))
+            .select((
+                &self.block_range,
+                &self.manifest_idx,
+                &self.param,
+                &self.context,
+                &self.causality_region,
+            ))
+            .order_by(&self.vid)
+            .load::<Tuple>(conn)?;
+
+        let mut count = 0;
+        for (block_range, src_manifest_idx, param, context, causality_region) in src_tuples {
```

Comment:
I suspect these N+1 queries may take longer than we'd like, but I don't have a better idea and N is most likely small for most subgraphs.